internal/espresso/config.go

```diff
@@ -0,0 +1,108 @@
+package espresso
+
+import (
+	"errors"
+	"fmt"
+	"github.com/rs/zerolog/log"
+	"github.com/saucelabs/saucectl/internal/config"
+	"gopkg.in/yaml.v2"
+	"os"
+	"strings"
+)
+
+// Project represents the espresso project configuration.
+type Project struct {
+	config.TypeDef `yaml:",inline"`
+	ShowConsoleLog bool
+	Sauce          config.SauceConfig `yaml:"sauce,omitempty" json:"sauce"`
+	Espresso       Espresso           `yaml:"espresso,omitempty" json:"espresso"`
+	Suites         []Suite            `yaml:"suites,omitempty" json:"suites"`
+	Artifacts      config.Artifacts   `yaml:"artifacts,omitempty" json:"artifacts"`
+}
+
+// Espresso represents espresso apps configuration.
+type Espresso struct {
+	App		string	`yaml:"app,omitempty" json:"app"`
+	TestApp	string	`yaml:"testApp,omitempty" json:"testApp"`
+}
+
+// TestOptions represents the espresso test filter options configuration.
+type TestOptions struct {
+	NotClass	[]string	`yaml:"notClass,omitempty" json:"notClass"`
+	Class	   	[]string	`yaml:"class,omitempty" json:"class"`
+	Package		string		`yaml:"package,omitempty" json:"package"`
+	Size		string		`yaml:"size,omitempty" json:"size"`
+	Annotation	string		`yaml:"annotation,omitempty" json:"annotation"`
+}
+
+// Suite represents the espresso test suite configuration.
+type Suite struct {
+	Name             string         	`yaml:"name,omitempty" json:"name"`
+	Devices          []config.Device	`yaml:"devices,omitempty" json:"devices"`
+	TestOptions      TestOptions		`yaml:"testOptions,omitempty" json:"testOptions"`
+}
+
+// Android constant
+const Android = "Android"
+
+// FromFile creates a new cypress Project based on the filepath cfgPath.
+func FromFile(cfgPath string) (Project, error) {
+	var p Project
+
+	f, err := os.Open(cfgPath)
+	defer f.Close()
+	if err != nil {
+		return Project{}, fmt.Errorf("failed to locate project config: %v", err)
+	}
+
+	if err = yaml.NewDecoder(f).Decode(&p); err != nil {
+		return Project{}, fmt.Errorf("failed to parse project config: %v", err)
+	}
+
+	if p.Sauce.Concurrency < 1 {
+		// Default concurrency is 2
+		p.Sauce.Concurrency = 2
+	}
+
+	return p, nil
+}
+
+// Validate validates basic configuration of the project and returns an error if any of the settings contain illegal
+// values. This is not an exhaustive operation and further validation should be performed both in the client and/or
+// server side depending on the workflow that is executed.
+func Validate(p Project) error {
+	if p.Espresso.App == "" {
```

Comment:
Should we also check that this ends with `.apk` ?

Reply:
Yeah, we can check the extension.